### DISCUSSION TITLE
File associations

### DISCUSSION THREAD

**Discussion by Keith94** [d2phap/ImageGlass#269] [General] [answered]
ImageGlass version: 4.5.11.14

I have some concerns/questions for the file associations tab, just in case there are any bugs ;)

1. The "Register extensions" button can be clicked even when no formats are checked.
1. Clicking on a format's name (to select it) unchecks all other formats
1. Formats don't stay checked permanently after they're associated in Windows
1. New file icons not being used with registered formats yet? (e6e4b67)

**Comment by d2phap** [comment, marked as answer]
Hi @Keith94 

> The "Register extensions" button can be clicked even when no formats are checked.

and

> Clicking on a format's name (to select it) unchecks all other formats

and

> Formats don't stay checked permanently after they're associated in Windows

The checkbox in the list is just to indicate selected item when you select. It doesn't mean the extensions you registered to system.

> New file icons not being used with registered formats yet? (e6e4b67)

With v4.5.11.14, you can use those new icons by clicking **Register Extensions** button. However, clicking **Register Extensions** button will just add ImageGlass to the list of Default app (#129) and tell the Windows the file format capabilities, it won't directly change the default app (because Windows restricted this). 
To associate the app to the extension, you can open **Default app settings** and select ImageGlass from the list. The icons only appear when you associate ImageGlass as default app for opening.

**Comment by Keith94** [comment]
Ahh, it makes sense now. I thought selecting them via the checkboxes was required, so I was confused by that odd behavior. Thank you for the explanation.
